cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamFetchHandler.java

```diff
@@ -107,9 +108,24 @@ public StreamFetchHandler(CConfiguration cConf, Authenticator authenticator,
   public void fetch(HttpRequest request, HttpResponder responder,
                     @PathParam("namespace-id") String namespaceId,
                     @PathParam("stream") String stream,
-                    @QueryParam("start") long startTime,
-                    @QueryParam("end") @DefaultValue("9223372036854775807") long endTime,
+                    @QueryParam("start") String start,
+                    @QueryParam("end") @DefaultValue("9223372036854775807") String end,
                     @QueryParam("limit") @DefaultValue("2147483647") int limit) throws Exception {
+    long startTime;
+    long endTime;
+    try {
+      /* if string is numeric then value is in milliseconds */
+      startTime = Long.parseLong(start);
```

Comment:
In TimerMathParser if it sees it's a timestamp that's numeric doesn't it assume that it it's in seconds (on line 93)?